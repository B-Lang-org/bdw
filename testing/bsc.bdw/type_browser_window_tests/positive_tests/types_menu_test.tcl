
# Creates a new project
handlers::open_project project.bspec 

# Cleans the build results
commands::full_clean

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the specified file
handlers::compile_file Mult4.bsv -withdeps

# Waits until the process with the 'BSPEC(BUILDPID)' id has finished
while {$BSPEC(BUILDPID) != ""} {
        update
}

# Loads the specified package
handlers::load_package Mult4

# Adds the specified type to the type hierarchy
handlers::add_type Tin

# Adds the specified type to the type hierarchy
handlers::remove_type Tin

# Cleans the build results
commands::full_clean

# Closes the current project
handlers::close_project

# Exits the workstation
exit
