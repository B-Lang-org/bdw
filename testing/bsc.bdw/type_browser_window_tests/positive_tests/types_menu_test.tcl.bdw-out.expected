Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult4.bsv
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult4.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult1.bsv
RESULT: code generation for mkMult1 starts
WARNING: Warning: "mult4/Mult1.bsv", line 8, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult1.ba
RESULT: compiling mult4/Mult2.bsv
RESULT: code generation for mkMult2 starts
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult2.ba
RESULT: compiling mult4/Mult4.bsv
RESULT: All packages are up to date.
Mult4 CompletionBuffer GetPut Connectable Vector RegFile Mult2 Mult1 Mult ListN Clocks Array List Inout FIFO FIFOF FIFOF_ Counter ConfigReg PreludeBSV Prelude
Tin

